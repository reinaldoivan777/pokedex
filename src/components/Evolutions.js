import React, { Fragment } from "react";
import { Link } from "react-router-dom";

export default function Evolutions({ datas }) {
  return (
    <Fragment>
      {datas.map(({ name, id, image, number }) => {
        return (
          <Link key={id} to={`/pokemon/${id}`}>
            <div className="text-left evolution-item">
              <img src={image} className="ml-3 image-evolution" alt={name} />
              <span className="mx-3">{`${number} --- ${name}`}</span>
            </div>
          </Link>
        );
      })}
    </Fragment>
  );
}
