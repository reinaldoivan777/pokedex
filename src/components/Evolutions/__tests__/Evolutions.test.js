import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import Evolutions from "../index";

describe("Evolutions", () => {
  it("render without error", () => {
    const datas = [
      {
        id: "UG9rZW1vbjowMDI=",
        name: "Ivysaur",
        number: "002",
        image: "https://img.pokemondb.net/artwork/ivysaur.jpg"
      },
      {
        id: "UG9rZW1vbjowMDM=",
        name: "Venusaur",
        number: "003",
        image: "https://img.pokemondb.net/artwork/venusaur.jpg"
      }
    ];
    renderer.create(
      <Router>
        <Evolutions datas={datas} />
      </Router>
    );
  });

  it("empty data", () => {
    renderer.create(
      <Router>
        <Evolutions datas={[]} />
      </Router>
    );
  });

  it("data null", () => {
    renderer.create(
      <Router>
        <Evolutions datas={null} />
      </Router>
    );
  });

  it("snapshot renders", () => {
    const datas = [
      {
        id: "UG9rZW1vbjowMDI=",
        name: "Ivysaur",
        number: "002",
        image: "https://img.pokemondb.net/artwork/ivysaur.jpg"
      },
      {
        id: "UG9rZW1vbjowMDM=",
        name: "Venusaur",
        number: "003",
        image: "https://img.pokemondb.net/artwork/venusaur.jpg"
      }
    ];
    const component = renderer.create(
      <Router>
        <Evolutions datas={datas} />
      </Router>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
