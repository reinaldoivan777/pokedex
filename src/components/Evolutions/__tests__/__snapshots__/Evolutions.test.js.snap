// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evolutions snapshot renders 1`] = `
Array [
  <a
    href="/pokemon/UG9rZW1vbjowMDI="
    onClick={[Function]}
  >
    <div
      className="text-left evolution-item"
    >
      <img
        alt="Ivysaur"
        className="ml-3 image-evolution"
        src="https://img.pokemondb.net/artwork/ivysaur.jpg"
      />
      <span
        className="mx-3"
      >
        002 --- Ivysaur
      </span>
    </div>
  </a>,
  <a
    href="/pokemon/UG9rZW1vbjowMDM="
    onClick={[Function]}
  >
    <div
      className="text-left evolution-item"
    >
      <img
        alt="Venusaur"
        className="ml-3 image-evolution"
        src="https://img.pokemondb.net/artwork/venusaur.jpg"
      />
      <span
        className="mx-3"
      >
        003 --- Venusaur
      </span>
    </div>
  </a>,
]
`;
